### DISCUSSION TITLE
Windows version?

### DISCUSSION THREAD

**Discussion by Drakemoor** [advplyr/audiobookshelf#235] [Ideas]
It looks fantastic; unfortunately, I have no idea how to use it since it has no installation for Windows, and I'm not familiar with those other OS.

Are you thinking of releasing a version for Windows?

Thanks,
M

**Comment by advplyr** [comment]
UPDATE: Looking for .NET developers to assist!

Yes, it will be released for Windows once an installer is made. I will post all updates to that here since this is the first thread on that topic.

Depending on how recent of a Windows version you are running you could install Docker and get the server up and running quickly until Windows binaries are released. I run the server on Windows all the time as a docker container.

**Reply by Drakemoor** [reply]
Thank you, @advplyr 

I'm glad and can't wait for the Windows version. I saw the Docker version, but my technical knowledge isn't sufficient to install and configure it. Besides, I'm not sure my computer configuration is powerful enough to support it. From what I read, Docker is a hefty beast.

Thanks again,
M

**Comment by Zachman1** [comment]
Are there further instructions on how to get this set up with docker? I just installed docker and tried running the commands that were given and ran into some errors.

The first command seemed to download some files but when I tried running the second part I got this

PS C:\Users\Zacha> docker run -d \
docker: invalid reference format.
See 'docker run --help'.
PS C:\Users\Zacha>     -e AUDIOBOOKSHELF_UID=99 \
-e : The term '-e' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:5
+     -e AUDIOBOOKSHELF_UID=99 \
+     ~~
    + CategoryInfo          : ObjectNotFound: (-e:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS C:\Users\Zacha>     -e AUDIOBOOKSHELF_GID=100 \
-e : The term '-e' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:5
+     -e AUDIOBOOKSHELF_GID=100 \
+     ~~
    + CategoryInfo          : ObjectNotFound: (-e:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS C:\Users\Zacha>     -p 13378:80 \
-p : The term '-p' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:5
+     -p 13378:80 \
+     ~~
    + CategoryInfo          : ObjectNotFound: (-p:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS C:\Users\Zacha>     -v </path/to/audiobooks>:/audiobooks \
At line:1 char:8
+     -v </path/to/audiobooks>:/audiobooks \
+        ~
The '<' operator is reserved for future use.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : RedirectionNotSupported

PS C:\Users\Zacha>     -v </path/to/config>:/config \
At line:1 char:8
+     -v </path/to/config>:/config \
+        ~
The '<' operator is reserved for future use.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : RedirectionNotSupported

PS C:\Users\Zacha>     -v </path/to/metadata>:/metadata \
At line:1 char:8
+     -v </path/to/metadata>:/metadata \
+        ~
The '<' operator is reserved for future use.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : RedirectionNotSupported

PS C:\Users\Zacha>     --name audiobookshelf \
At line:1 char:7
+     --name audiobookshelf \
+       ~
Missing expression after unary operator '--'.
At line:1 char:7
+     --name audiobookshelf \
+       ~~~~
Unexpected token 'name' in expression or statement.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : MissingExpressionAfterOperator


I am sure I am doing something basic wrong, but do not know enough about Docker to fix it.

**Reply by advplyr** [reply]
Change the command to be on a single line. Remove the \\'s

**Reply by Zachman1** [reply]
I tried this and still got an error

PS C:\Users\Zacha> docker pull advplyr/audiobookshelf docker run -d  -e AUDIOBOOKSHELF_UID=99  -e AUDIOBOOKSHELF_GID=100  -p 13378:80  -v I:\A:/audiobooks  -v I:\C:/config  -v I:\M:/metadata  --name audiobookshelf  ghcr.io/advplyr/audiobookshelf
unknown shorthand flag: 'd' in -d

**Reply by advplyr** [reply]
docker pull and docker run are 2 separate commands

**Reply by Zachman1** [reply]
Thank you. Than ran and gave me a string of letters and numbers and I see the folders are now populated. 

Is a gui supposed to pop up?

Thank you for your help so far.

**Reply by advplyr** [reply]
When the container is running you should be able to access the GUI from your browser by visiting the ip address of where it is running.
If you are running it on the same computer that you are using you should be able to access it by going to `localhost:13378` in your browser.
If it is a different computer on the same network then you will most likely use an ip address like `192.168.0.x:13378` where x is specific to the device on your network.

**Reply by Zachman1** [reply]
I got it working and on my phone! Thank you so much for the help.

Perhaps I should make a seperate post for this, but I am able to stream audiobooks but when I attempt to download I get the error "Item download Complete but failed to create library Item" The download goes from 0% to 100% right away and nothing is downloaded. I  have given the app file permissions and tried a few different areas in my storage. This  is on the android app.

**Reply by advplyr** [reply]
There are a few issues opened for this on the audiobookshelf-app github repo. One issue might be that you have a semicolon in your book title, that issue will be fixed in the next release. Another issue that could be worth checking is to use a folder that is already on your phone by default (like "Audiobooks" or "Downloads").
Still working out the kinks with the new update to the app, you can follow along all the issues on that github too.

**Reply by Zachman1** [reply]
Perfect. I seem to have everything working as it should be. Again thank you for the help. I really like the program and app so far.

**Comment by cyturralde** [comment]
Just wanted to add another vote for a Windows version.

**Comment by TheRealRws** [comment]
+1 on this. Already have stuff running on a windows server would love to see this available for it!
